src/codecovio.jl

```diff
@@ -140,21 +140,27 @@ module Codecov
     end
 
 
-    import Git
+    import Base.LibGit2
 
     """
-        submit_local(fcs::Vector{FileCoverage})
+        submit_local(fcs::Vector{FileCoverage}, dir::AbstractString=pwd())
 
-    Takes a `Vector` of file coverage results (produced by `process_folder`),
-    and submits them to Codecov.io. Assumes the submission is being made from
-    a local git installation.  A repository token should be specified by a
-    'token' keyword argument or the `CODECOV_TOKEN` environment variable.
+    Take a `Vector` of file coverage results (produced by `process_folder`),
+    and submit them to Codecov.io. Assumes the submission is being made from
+    a local git installation, rooted at `dir`. A repository token should be specified by a
+    `token` keyword argument or the `CODECOV_TOKEN` environment variable.
     """
-    function submit_local(fcs::Vector{FileCoverage}; kwargs...)
-        kwargs = set_defaults(kwargs,
-            commit = Git.readchomp(`rev-parse HEAD`, dir=""),
-            branch = Git.branch(dir="")
-            )
+    function submit_local(fcs::Vector{FileCoverage}, dir::AbstractString=joinpath(pwd(),".."); kwargs...)
+        LibGit2.with(LibGit2.GitRepo(dir)) do repo
+            LibGit2.with(LibGit2.head(repo)) do headref
+                branch_name = LibGit2.shortname(headref)
+                commit_oid  = LibGit2.GitHash(LibGit2.peel(headref))
+                kwargs = set_defaults(kwargs,
+                    commit = string(commit_oid),
+                    branch = LibGit2.branch(headref) # this function returns a String
```

Comment:
This appears to be triggering the error in the PR build: `ERROR: LoadError: GitError(Code:ERROR, Class:Invalid, Reference 'HEAD' is neither a local nor a remote branch.)` I wonder if it's because Travis checks out the merge commit for PR builds.

Reply:
Interesting...

Reply:
I could be wrong, but according to the documentation for the LibGit2 module, `LibGit2.branch` _creates_ a new branch. If you look at the definition of `Git.branch` from the Git package, which was used previously, it appears to be getting the name of the branch currently being built: https://github.com/JuliaPackaging/Git.jl/blob/master/src/Git.jl#L165.

Reply:
yeaaaaaah oh no i did this wrong will fix

Reply:
WAIT